test/functional/specs/desktop/C14394.js

```diff
@@ -0,0 +1,62 @@
+import { t, Selector, ClientFunction } from "testcafe";
+import createNetworkLogger from "../../helpers/networkLogger";
+import getResponseBody from "../../helpers/networkLogger/getResponseBody";
+import { responseStatus } from "../../helpers/assertions/index";
+import fixtureFactory from "../../helpers/fixtureFactory";
+import testServerUrl from "../../helpers/constants/testServerUrl";
+
+const urlCollector = `${testServerUrl}/test/functional/sandbox/html/alloySdk.html`;
+const networkLogger = createNetworkLogger();
+
+fixtureFactory({
+  title:
+    "C14394: When ID migration is enabled and no identity cookie is found but legacy AMCV cookie is found, the ECID will be sent on the request",
+  url: urlCollector,
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C14394",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const apiCalls = ClientFunction(() => {
+  document.cookie =
+    "AMCV_53A16ACB5CC1D3760A495C99%40AdobeOrg=77933605%7CMCIDTS%7C18290%7CMCMID%7C16908443662402872073525706953453086963%7CMCAAMLH-1580857889%7C9%7CMCAAMB-1580857889%7CRKhpRz8krg2tLO6pguXWp5olkAcUniQYPHaMWWgdJ3xzPWQmdj0y%7CMCOPTOUT-1580260289s%7CNONE%7CvVersion%7C4.5.1";
+
+  window.alloy("configure", {
+    configId: "9999999",
+    orgId: "53A16ACB5CC1D3760A495C99@AdobeOrg",
+    edgeBasePath: window.edgeBasePath,
+    debugEnabled: true,
+    idMigrationEnabled: true
+  });
+});
+
+test("Test C14394: When ID migration is enabled and no identity cookie is found but legacy AMCV cookie is found, the ECID will be sent on the request", async () => {
+  await apiCalls();
+
+  // NOTE: I assume a click is needed to get a response body. Putting the same event call in apiCalls() results in an undefined response body.
```

Comment:
When you say "same event call", can you show me what the event call you were trying to put inside `apiCalls` looks like?

Reply:
Are you talking about:

```
alloy("event", {
  viewStart: true,
  // TODO: Change data to match final XDM schema.
  xdm: {
    key: "value"
  }
}
```

Reply:
I wasn't returning the event call promise in t.eval(). That was the problem.